base/uk.ac.stfc.isis.ibex.ui.graphing/src/uk/ac/stfc/isis/ibex/ui/graphing/websocketview/MatplotlibWebsocketModel.java

```diff
@@ -177,6 +180,26 @@ private void setConnectionStatus(final boolean isConnected) {
 		viewModel.onConnectionStatus(isConnected);
 	}
 	
+	/**
+	 * Notifies the server that the plot is to be navigated, force refresh
+	 * the server 
+	 * @param name the button's navigation type
+	 */
+	public void navigatePlot(String name) {
+		workerThread.submit(() -> connection.navigatePlot(name));
+		workerThread.submit(() -> connection.forceServerRefresh());
+	}
+	
+	/**
+	 * Notifies the server that a mouse button as been pressed/released over the figure 
+	 * @param position the cursor position
+	 * @param pressType the button event type
+	 */
+	public void notifyButtonPress(final MatplotlibCursorPosition position, String pressType) {
+		workerThread.submit(() -> connection.notifyButtonPress(position, pressType));
+		workerThread.submit(() -> connection.forceServerRefresh());
```

Comment:
```suggestion
		workerThread.submit(() -> {
		    connection.notifyButtonPress(position, pressType);
		    connection.forceServerRefresh()
		});
```

& similar for other locations